Add DisableRationaleRule

acecilia opened the pull request:
Solves https://github.com/realm/SwiftLint/issues/2655

acecilia commented on Tests/SwiftLintFrameworkTests/RulesTests.swift:
```diff
@@ -2,6 +2,15 @@ import SwiftLintFramework
+        verifyRule(
+            DisableRationaleRule.description,
+            commentDoesntViolate: false,
+            skipStringTests: true,
```
I disabled the string tests because, even though I used `let disableMatches = file.match(pattern: "swiftlint:disable", with: [.comment])` (note the `.comment`), still violations are raised when executing the string tests. 
I have been looking into it but could not find the reason behind it, could it be a bug inside `let disableMatches = file.match(pattern: "swiftlint:disable", with: [.comment])`?

acecilia replied:
Okey, I think the problem was that my code has double quote characters in it that where not being escaped when converting them to string literal. I think I fixed that, now string tests are also passing.

acecilia commented:
Also, I tried to implement the rule using `ASTRule` (this is my first time doing a Swiftlint rule), but seems to me that it is not possible to get comment information from the AST at the moment, right? Only detecting `SwiftDeclarationKind` and `SwiftExpressionKind` work.

Dschee commented on Source/SwiftLintFramework/Rules/Idiomatic/DisableRationaleRule.swift:
```diff
@@ -0,0 +1,144 @@
+
+    public func validate(file: File) -> [StyleViolation] {
+        let disableMatches = file.match(pattern: "swiftlint:disable", with: [.comment])
+        guard disableMatches.isEmpty == false else {
```
The ` == false` should be removed here, use `!` instead.

acecilia commented:
@Dschee Thanks! About the (non-)triggering examples:

1. multiple rules are disabled in a single line: I will be adding them.
2. the disable comment is not done on it's own line but on a code line: this is already done. See [here](https://github.com/realm/SwiftLint/pull/2660/files#diff-b5d5ebb28e58c901be755c56894b349bR3080) and [here](https://github.com/realm/SwiftLint/pull/2660/files#diff-b5d5ebb28e58c901be755c56894b349bR3084) for non-triggering, and [here](https://github.com/realm/SwiftLint/pull/2660/files#diff-b5d5ebb28e58c901be755c56894b349bR3129) for triggering.

I will be addressing your comments about rule direction in the other issue.
